### ISSUE TITLE
Nested sorting 

### ISSUE THREAD

**Issue by ValentinVignal** [TriPSs/nestjs-query#194]
**Is your feature request related to a problem? Please describe.**

There were some issues on https://github.com/doug-martin/nestjs-query about being able to sort by nested fields:
- https://github.com/doug-martin/nestjs-query/issues/559
- https://github.com/doug-martin/nestjs-query/issues/752

which could be super helpful. I would like to be able to sort a list of Todos by their linked project name:

```graphql
query myTodos {
  todos(
    # Sort by project name here
  ) {
    id # ID!
    name  # String!
    completed # Boolean!
    project {  # Project!
      id # ID!
      name # String!
    }
  }
}
```

In my case, I only have the need for "one-to-one" or "one-to-many" relationships (1 todo won't be linked to multiple projects)

### Have you read the [Contributing Guidelines](https://github.com/tripss/nestjs-query/blob/master/CONTRIBUTING.md)?

Yes

**Describe the solution you'd like**

The solution offered https://github.com/doug-martin/nestjs-query/issues/559#issuecomment-713290342 was not too bad:

```graphql
TodoSortFields {
  id
  name
  project__id
  project__name
}
```

**Describe alternatives you've considered**
Writing my own custom resolver where I can implement my own sorting.

**Comment by dvins** [comment]
Separating the table from the column using  a `.` might be better syntax.

**Comment by ValentinVignal** [comment]
> Separating the table from the column using a `.` might be better syntax.

I don't think that possible, see this comment https://github.com/doug-martin/nestjs-query/issues/559#issuecomment-713290342

**Comment by dvins** [comment]
Thanks for pointing out @ValentinVignal, the double underscore makes sense.

**Comment by dvins** [comment]
By chance @TriPSs have you been able to look at this or have any insights on a preferred approach?

**Comment by TriPSs** [comment]
Have not looked at it, `__` think it's good, changes of that colliding with other fields should be pretty low.

**Comment by Justin99b** [comment]
Any updates on this?

**Comment by TriPSs** [comment]
Closing as part of repository cleanup. If this issue is still relevant, please reopen.
